lib/grplatt.gi

```diff
@@ -3335,57 +3335,254 @@ local c,n,deg,ind,core,i,j,sum;
+      m:=MaximalSubgroupClassReps(sub:cheap:=false);
+    fi;
+
+    # remove duplicates 
+    m:=Filtered(m,x->ForAll(l,y->Size(x)<>Size(y) or
+        ForAny(GeneratorsOfGroup(x),z->not z in y)));
+
+    if bound>1 then
+      if not IsPerfectGroup(sub) then
+        a:=BoundedIndexAbelianized(G,sub,bound);
+        if Size(a)<Size(sub) then
+          len:=Length(m);
+          m:=Filtered(m,x->not IsSubset(x,a));
+#         Print("Dropped ",len-Length(m)," by abelian\n");
+          Add(m,a);
+        fi;
+      fi;
+      # do we already have a bit smaller?
+      m:=Filtered(m,x->ForAll(l,y->Size(x)>=bound*Size(y) or
+          ForAny(GeneratorsOfGroup(y),z->not z in x)));
+
+      sz:=List(Filtered(Collected(List(m,Size)),x->x[2]>1),x->x[1]);
+      for i in sz do
+        len:=Length(m);
+        a:=Filtered(m,x->Size(x)=i);
+        m:=Filtered(m,x->Size(x)<>i);
+        # now try intersections
+        tb:=infinity;
+        while tb=infinity do
+          for j in [1..Length(a)] do
+            for k in [1..j-1] do
+              if IsBound(a[j]) and IsBound(a[k]) 
+                and Size(a[j])*bound>=2*i and Size(a[k])*bound>=2*i then
+                b:=Intersection(a[j],a[k]);
+                if Size(b)*bound>=i then
+                  Unbind(a[j]);
+                  for r in [1..Length(a)] do
+                    if IsBound(a[r]) and IsSubset(a[r],b) then
+                      Unbind(a[r]);
+                    fi;
+                  od;
+                  a[k]:=b;
+                fi;
+                tb:=Minimum(tb,i/Size(b));
+              fi;
+            od;
+          od;
+          a:=Filtered(a,x->IsBound(x));
+          if tb>bound and Length(a)>5*10^(1+LogInt((1+QuoInt(tb,bound)),prime)) then
+            bound:=bound*prime;
+            tb:=infinity;
+          fi;
+        od;
+
+        m:=Concatenation(m,a);
+#        Print("Dropped ",len-Length(m)," to ",Length(a)," by size ",i,"\n");
+      od;
+    fi;
+    return m;
+  end;
+
   return IteratorByFunctions(rec(NextIterator:=function(iter)
-             local a;
-              if pos>Length(l) then
-                if mode=2 then return fail;
-                elif mode=1 and max^2>Size(G) and lev>=3 then
-                  mode:=2;
+             local a,b,m,i,j;
+              
+              if Length(l)=0 then
+                # no groups there. Start getting new ones
+                a:=Filtered(process,x->Size(x)>=divs[Length(divs)]);
+                process:=Filtered(process,x->Size(x)<divs[Length(divs)]);
+                for j in a do
+                  m:=maxer(j);
+                  Append(l,m);
+                od;
+                SortBy(l,Size);
+              fi;
+
+              if Size(l[Length(l)])<divs[Length(divs)] then
+                # new size. 
+
+                if Length(process)>0 
+                 and Size(process[Length(process)])<=limit then
+                  # switch to lattice
+                  a:=Size(l[Length(l)]);
```

Comment:
Observation: The code in this `if` statement is never executed by the Travis test suite.

Reply:
Yes. Once subgroups of size <=limit are required, it is worth top do the whole lattice rather than do iterated maximals. I have not been able to find an example of this case that is small enough that I would not have to worry about the test not timing out. (I have run this in larger examples by hand.)